src/ServiceControl.Audit.Persistence.RavenDB/DatabaseSetup.cs

```diff
@@ -1,69 +1,74 @@
 ﻿namespace ServiceControl.Audit.Persistence.RavenDB
 {
+    using System;
     using System.Collections.Generic;
     using System.Threading;
     using System.Threading.Tasks;
     using Raven.Client.Documents;
     using Raven.Client.Documents.Indexes;
-    using Raven.Client.Documents.Operations;
     using Raven.Client.Documents.Operations.Expiration;
     using Raven.Client.Documents.Operations.Indexes;
     using Raven.Client.Exceptions;
-    using Raven.Client.Exceptions.Database;
     using Raven.Client.ServerWide;
     using Raven.Client.ServerWide.Operations;
+    using Raven.Client.ServerWide.Operations.Configuration;
     using ServiceControl.Audit.Persistence.RavenDB.Indexes;
     using ServiceControl.SagaAudit;
 
     class DatabaseSetup(DatabaseConfiguration configuration)
     {
         public async Task Execute(IDocumentStore documentStore, CancellationToken cancellationToken)
         {
-            try
-            {
-                await documentStore.Maintenance.ForDatabase(configuration.Name).SendAsync(new GetStatisticsOperation(), cancellationToken);
-            }
-            catch (DatabaseDoesNotExistException)
+            await CreateDatabase(documentStore, configuration.Name, cancellationToken);
+            await UpdateDatabaseSettings(documentStore, configuration.Name, cancellationToken);
+
+            await DeleteLegacySagaDetailsIndex(documentStore, cancellationToken);
+            await CreateIndexes(documentStore, cancellationToken);
+
+            await ConfigureExpiration(documentStore, cancellationToken);
```

Comment:
Does this have immediate effect, and not require a database reset that UpdateDatabaseSettings() is doing as well? (Same for Primary below)